src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1644,14 +1644,31 @@ private static TypeSymbol GetParameterType(int argIndex, MemberAnalysisResult re
                 }
             }
 
-            // Finally, if one has fewer custom modifiers, that is better
+            // Otherwise, if one has fewer custom modifiers, that is better
             int m1ModifierCount = m1.LeastOverriddenMember.CustomModifierCount();
             int m2ModifierCount = m2.LeastOverriddenMember.CustomModifierCount();
             if (m1ModifierCount != m2ModifierCount)
             {
                 return (m1ModifierCount < m2ModifierCount) ? BetterResult.Left : BetterResult.Right;
             }
 
+            // Otherwise, prefer methods with 'val' parameters over 'in' parameters.
+            Debug.Assert(m1Original.Length == m2Original.Length, "Unequal parameters counts are handled in an earlier stage");
```

Comment:
>"Unequal parameters counts are handled in an earlier stage" [](start = 65, length = 59)

Could you please point to the place that does that?

Reply:
Check lines 1441-1519. They should handle these cases where we end up with mismatching parameter count because of unused optional or expanded ones. I'll add a test as well.

---
In reply to: [151523306](https://github.com/dotnet/roslyn/pull/23122#discussion_r151523306) [](ancestors = 151523306)

Reply:
I see that code comparing the number of parameters used, but not the number of parameters declared.

---
In reply to: [151544214](https://github.com/dotnet/roslyn/pull/23122#discussion_r151544214) [](ancestors = 151544214,151523306)